addons/mail/static/src/core/common/link_preview.js

```diff
@@ -44,4 +44,8 @@ export class LinkPreview extends Component {
     onImageLoaded() {
         this.env.onImageLoaded?.();
     }
+
+    get canDelete() {
+        return this.env.message?.allowsEdition;
+    }
```

Comment:
Would put this immediately in template, to avoid defining an indirection level with getter.

I get that `LinkPreview` component, by its name, should not depend on message, but it does and should actually be named `MessageLinkPreview`. List of props would be trivially `messagePreviewList` record and code would be easier to understand if the component was renamed.